### PR TITLE
add action operator implement and crd implement

### PR DESCRIPTION
**Pull request by cdjianghan**
add action operator implement and crd implement
add app crd and app operator structure

### CODE REVIEW

File: controllers/app/api/v1/actions_types.go [hunk 1]
```diff
@@ -20,34 +20,32 @@ import (
...
-	// Application name, actions will bind to an application
-	AppName string `json:"appName,omitempty"`
-	// Alias will show it on dashboard. like DELETE UPDATE MIGRATE...
-	Alias string `json:"alias,omitempty"`
-	// Args, the actions input args, like oss backup address...
-	Args map[string]string `json:"args,omitempty"`
-	// Actions exec command, support render some args and downward API, like application name, service name ...
-	CMD string `json:"cmd,omitempty"`
+	AppName    string            `json:"appName,omitempty"`
+	ActionName string            `json:"actionName,omitempty"`
```
Comment (fanux) [inline controllers/app/api/v1/actions_types.go:38]: What is the difference with meta.Name？

Comment (cdjianghan) [reply controllers/app/api/v1/actions_types.go:38]: eeee,that is your code ,  I forgot to move it . I'll fix it

Comment (cdjianghan) [reply controllers/app/api/v1/actions_types.go:38]: no difference

File: controllers/app/controllers/actions_controller.go [hunk 2]
```diff
@@ -17,46 +17,139 @@ limitations under the License.
...
+	}
 
-	// TODO(user): your logic here
+	image := &imagehubv1.Image{}
+	if err := r.Get(ctx, client.ObjectKey{Namespace: DefaultNameSpace, Name: action.Spec.AppName}, image); err != nil {
+		r.recorder.Eventf(image, core.EventTypeNormal, "ImageGetFailed", "Infra %s status is pending", image.Name)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	r.Logger.Info(fmt.Sprintf("image.Name: %v", image.Name))
+	switch image.Spec.DetailInfo.Actions[image.Name].ActionType {
```
Comment (fanux) [inline controllers/app/controllers/actions_controller.go:80]: engine := NewEngine(type)

The caller don't care about use what engine.

File: controllers/app/controllers/actions_controller.go [hunk 3]
```diff
@@ -17,46 +17,139 @@ limitations under the License.
...
 
-// SetupWithManager sets up the controller with the Manager.
 func (r *ActionsReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.recorder = mgr.GetEventRecorderFor("sealos-action-controller")
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&appv1.Actions{}).
 		Complete(r)
 }
+
+func (ctlEngine *KubectlEngine) Parse() error {
```
Comment (fanux) [inline controllers/app/controllers/actions_controller.go:118]: This implementation is more suitable to be placed in another file.

File: controllers/app/controllers/actions_controller.go [hunk 4]
```diff
@@ -17,46 +17,139 @@ limitations under the License.
...
+	var byt bytes.Buffer
+	e := t.Execute(&byt, args)
+	if e != nil {
+		return fmt.Errorf("action template execute failed")
+	}
+	ctlEngine.parseResult = byt.Bytes()
+	return nil
+}
+
+func (ctlEngine *KubectlEngine) Apply() error {
```
Comment (fanux) [inline controllers/app/controllers/actions_controller.go:136]: If actions are executed with administrator privileges, there will be security issues

**Comment by fanux** [comment]
Uniform lowercase file names

**Review by fanux — APPROVED** [review]

**Pull request merged**